### ISSUE TITLE
Ever-growing memory usage

### ISSUE THREAD

**Issue by imbrem** [nvarner/typst-lsp#66]
- [x] I have searched the [issues](https://github.com/nvarner/typst-lsp/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [discussions](https://github.com/nvarner/typst-lsp/discussions) and believe that my question is not already covered.

## Feature Request

The memory usage of the typst-lsp grows linearly with time, until after a few hours of intense usage it reaches nearly 32 GB of RAM. Most of this seems to be unused entries in typst's comemo cache, since swapping doesn't hurt performance much. Could the LSP be configured to perform regular evictions of old cache entries, via https://docs.rs/comemo/latest/comemo/fn.evict.html?

**Comment by nvarner** [comment]
Thank you for the report. I'm currently working on a major refactoring to address some foundational issues that will get harder to resolve as the project grows. I'm focusing my development time on that because it blocks everything else, as further changes may reverse some of the progress I'm making. In particular, I plan to finish the refactor before writing code to address this.

However, this is a major problem. If you or anyone else submits a PR for it, I'll merge it when it comes in. Otherwise, it will be next on my to-do list, and should be fixed within the next week, barring unforeseen difficulties.

**Comment by nvarner** [comment]
Does restarting the editor work as a temporary fix?

**Comment by laurmaedje** [comment]
Should be fixed once #71 is merged.

**Comment by nvarner** [comment]
#71 was just merged. I'll release a new version soon, so you'll be able to update the extension and everything should work.

**Comment by nvarner** [comment]
The fix is now live. Try updating the extension, and reopen the issue if memory grows huge again.

**Comment by manuel5975p** [comment]
@nvarner This is still a problem for me, and I have 64 GiB of ram. Should I open a new issue?
